proxy/ProxyClientSession.h

```diff
@@ -30,6 +30,8 @@
 #include "InkAPIInternal.h"
 #include "http/HttpServerSession.h"
 
+extern volatile bool http_client_session_draining;
```

Comment:
Can I left that as a future work? Because main purpose of this PR is generalizing the flag and supporting graceful shutdown on HTTP/1.1. This is just a replacement of `http2_drain`.